src/app/features/activity-list/components/transactions/transactions-list.tsx

```diff
@@ -0,0 +1,49 @@
+import { useMemo } from 'react';
+
+import { BitcoinTransaction as BitcoinTx } from '@app/common/transactions/bitcoin/transaction.models';
+import { whenSupportedTransaction } from '@app/common/transactions/supported-transactions';
+import { StacksTransactions } from '@app/common/transactions/stacks/transaction.models';
+import {
+  getTransactionId,
+  getTransactionType,
+} from '@app/common/transactions/supported-transactions.utils';
+
+import { TransactionsByDateLayout } from './transactions-by-date.layout';
+import { TransactionsListLayout } from './transactions-list.layout';
+import { createTxDateFormatList } from './transactions.utils';
+import { StacksTransaction } from './stacks-transaction';
+import { BitcoinTransaction } from './bitcoin-transaction';
+
+interface TransactionsListProps {
+  bitcoinTxs: BitcoinTx[];
+  stacksTxs: StacksTransactions[];
+}
+export function TransactionsList({ bitcoinTxs, stacksTxs }: TransactionsListProps) {
+  const txsGroupedByDate = useMemo(
+    () =>
+      bitcoinTxs.length || stacksTxs.length ? createTxDateFormatList(bitcoinTxs, stacksTxs) : [],
+    [bitcoinTxs, stacksTxs]
+  );
+
+  return (
+    <TransactionsListLayout>
+      {txsGroupedByDate.map(({ date, displayDate, txs }) => (
+        <TransactionsByDateLayout date={date} displayDate={displayDate}>
+          {txs.map(tx => {
+            return whenSupportedTransaction(getTransactionType(tx))({
+              bitcoin: (
+                <BitcoinTransaction key={getTransactionId(tx)} transaction={tx as BitcoinTx} />
+              ),
+              stacks: (
+                <StacksTransaction
+                  key={getTransactionId(tx)}
+                  transaction={tx as StacksTransactions}
+                />
+              ),
+            });
+          })}
```

Comment:
This component is pretty clean as is, but we could also have a generic `<TransactionListItem />` that takes care of this routing